include/natalie_parser/node.hpp

```diff
@@ -1329,6 +1331,20 @@ class SplatNode : public Node {
     Node *m_node { nullptr };
 };
 
+class SplatValueNode : public Node {
+public:
+    SplatValueNode(const Token &token, Node *value)
+        : Node { token }
+        , m_value { value } { }
```

Comment:
You also need a destructor like this one: https://github.com/seven1m/natalie/blob/eafe3045f7bafc096d77bd8de12216dfdf2533db/include/natalie_parser/node.hpp#L173-L175